### PR TITLE
Change the base image to debian-base

### PR DESCRIPTION
**Pull request by vpnachev**
**Is this a bug fix or adding new feature?**
I would say it is a security improvement.

When amazonlinux is used as base image we are detecting multiple critical vulnerabilities in the image, some of them are 
1. CVE-2016-4658
1. CVE-2017-7376
1. CVE-2016-4448
1. CVE-2016-1834
1. CVE-2017-16997
1. CVE-2017-18269
1. CVE-2019-5482
1. CVE-2019-5481
1. CVE-2019-3822
...
And many others.

**What is this PR about? / Why do we need it?**
Change the base image to ~alpine linux~ debian-base-amd64 to reduce the number of reported security vulnerabilities and also the size of the image from:
```
$ docker image ls amazon/aws-ebs-csi-driver:latest
REPOSITORY                  TAG                 IMAGE ID            CREATED             SIZE
amazon/aws-ebs-csi-driver   latest              ddab7a7bfa60        3 seconds ago       464MB # amazonlinux as base image
```
to
```
$ docker image ls amazon/aws-ebs-csi-driver:latest
REPOSITORY                  TAG                 IMAGE ID            CREATED             SIZE
amazon/aws-ebs-csi-driver   latest              17fafe83d963        19 seconds ago      36.7MB # alpinelinux as base image
```

```
$  docker image ls amazon/aws-ebs-csi-driver:latest
REPOSITORY                  TAG                 IMAGE ID            CREATED             SIZE
amazon/aws-ebs-csi-driver   latest              f2432a80d7ef        2 seconds ago       167MB # debian-base as base image
```
**What testing is done?**

### CODE REVIEW

**Comment by bertinatto** [comment]
Thank you for your PR, @vpnachev.

This is a really good point. @leakingtapan @matthew are you OK with that?

Even more interesting would be to have distroless as the base image. There's an issue for that here: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/272. A this has been considered in the PD driver as well: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/177.

**Comment by msau42** [comment]
Ref discussion why K8s chose debian over alpine as a base image: https://github.com/kubernetes/kubernetes/issues/40248#issuecomment-280781931

I'm actually using debian-slim as the base image, and it dropped the image size to 48mb

**Comment by danbeaulieu** [comment]
+1 to distroless if the base layer needs to change. In the cloud native space alpine is an outlier since many projects have adopted slim images or distroless. Adding another non-standard distro to infrastructure complicates container supply chain management.

**Comment by vpnachev** [comment]
/retest

Switching to distroless would require a major refactoring in the build system, i.e. to be switched to bazel, because of the dependencies like `e2fsprogs`, `xfsprogs` and `util-linux` which is not the focus of this PR.

What I can do is to use use the same image as for the GCP CSI driver https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/e1670c165e72bca32c778b884656b03ff1818386/Dockerfile#L25. What do you think?

**Comment by vpnachev** [comment]
Actually, the more widely used base image for CSI components is us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0.

**Comment by wongma7** [comment]
I would prefer to maintain both a Dockerfile.amazonlinux (amazon still need an AL2 based image) and a debian one.

**Comment by vpnachev** [comment]
Dockerfile.amazonlinux has been added.

**Comment by vpnachev** [comment]
A mitigation for https://github.com/fvbommel/util/issues/6 has been implemented.

File: Dockerfile [hunk 1]
```diff
@@ -23,8 +23,8 @@ RUN go mod download
 ADD . .
 RUN make
 
-FROM amazonlinux:2
-RUN yum install ca-certificates e2fsprogs xfsprogs util-linux -y
+FROM us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0
```
Comment (leakingtapan) [inline Dockerfile:26]: is there a multi arch image for this? I would prefer stick with multi-arch image to make multi arch image build easier.

File: go.mod [hunk 2]
```diff
@@ -51,6 +49,7 @@ replace (
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.17.3
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.17.3
 	k8s.io/sample-controller => k8s.io/sample-controller v0.17.3
+	vbom.ml/util => github.com/fvbommel/util v0.0.2 // Mitigate https://github.com/fvbommel/util/issues/6
```
Comment (leakingtapan) [inline go.mod:52]: What is this dependency for?

Comment (gnufied) [reply go.mod:52]: It is being pulled via k8s dependency chain.  Unit tests are failing atm in this repo without the fix:

```
Activated service account credentials for: [pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com]
+ make test
go test -v -race ./cmd/... ./pkg/...
go: k8s.io/kubernetes@v1.17.3 requires
	k8s.io/kubectl@v0.0.0 requires
	vbom.ml/util@v0.0.0-20160121211510-db5cfe13f5cc: unrecognized import path "vbom.ml/util": https fetch: Get "https://vbom.ml/util?go-get=1": dial tcp: lookup vbom.ml on 10.63.240.10:53: no such host
make: *** [Makefile:67: test] Error 1
+ EXIT_VALUE=2
+ set +o xtrace
```

Comment (vpnachev) [reply go.mod:52]: Ref: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/551#issuecomment-682387518

File: Dockerfile (hunk 1 above)
Comment (ialidzhikov) [reply Dockerfile:26]: `k8s.gcr.io/build-image/debian-base` (the same image but using the vanity domain) is multi arch.

```suggestion
FROM k8s.gcr.io/build-image/debian-base:v2.1.3
```

Comment (vpnachev) [reply Dockerfile]: Thanks. I have done it.

**Comment by vpnachev** [comment]
/test pull-aws-ebs-csi-driver-e2e-single-az

However, I think the failure of this test is related to https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/545

**Comment by ialidzhikov** [comment]
@leakingtapan , @bertinatto did you managed to recheck this PR after the recent changes?

**Comment by wongma7** [comment]
@vpnachev looks like Dockerfile.amazonlinux got dropped. There is some red tape preventing me from pushing a non-AL2 based image to the official amazon Docker Hub https://hub.docker.com/r/amazon/aws-ebs-csi-driver. So for the next release which I want to make this week, we might still need to base it on AL2. What I can do is partially revert this, make the release, then resubmit it and try to get an exception to push debian-based image to hub.docker.com/u/amazon in time for the next release. WDYT?

**Comment by wongma7** [comment]
Actually I guess all I need to do is change the github action to build with target al2, I see now your Dockerfile allows for that. Thanks!!

**Pull request merged**